src/commands.ts

```diff
@@ -383,36 +387,39 @@ async function openWorkspace(
     newWindow = false
   }
 
-  // If a folder isn't specified, we can try to open a recently opened folder.
-  if (!folderPath) {
+  // If a folder isn't specified or we have been asked to open the most recent,
+  // we can try to open a recently opened folder/workspace.
+  if (!folderPath || openRecent) {
     const output: {
       workspaces: { folderUri: vscode.Uri; remoteAuthority: string }[]
     } = await vscode.commands.executeCommand("_workbench.getRecentlyOpened")
     const opened = output.workspaces.filter(
-      // Filter out `/` since that's added below.
+      // Remove recents that do not belong to this connection.  The remote
+      // authority maps to a workspace or workspace/agent combination (using the
+      // SSH host name).  This means, at the moment, you can have a different
+      // set of recents for a workspace versus workspace/agent combination, even
+      // if that agent is the default for the workspace.
       (opened) => opened.folderUri?.authority === remoteAuthority,
     )
-    if (opened.length > 0) {
-      let selected: (typeof opened)[0]
 
-      if (opened.length > 1) {
-        const items: vscode.QuickPickItem[] = opened.map((folder): vscode.QuickPickItem => {
-          return {
-            label: folder.folderUri.path,
-          }
-        })
-        const item = await vscode.window.showQuickPick(items, {
-          title: "Select a recently opened folder",
-        })
-        if (!item) {
-          return
+    // openRecent will always use the most recent.  Otherwise, if there are
+    // multiple we ask the user which to use.
+    if (opened.length === 1 || (opened.length > 1 && openRecent)) {
+      folderPath = opened[0].folderUri.path
+    } else if (opened.length > 1) {
+      const items: vscode.QuickPickItem[] = opened.map((folder): vscode.QuickPickItem => {
```

Comment:
Not a huge deal, but maybe it could be helpful to pass the type parameter into the `.map` call to remove the need to type the array and the callback output?

```ts
const items = opened.map<vscode.QuickPickItem>((folder)  => {
  // Logic goes here
});
```

Reply:
Oo yeah I like this.